// 需要与客户端共享的类型，常量等

syntax = "proto3";
package rpc_server;

// 服务器类型
enum ServerType
{
    UNKNOWN = 0;		  // 未知
    BATTLE = 1;           // 战斗服务器  
    CENTRAL = 2;          // 中心服务器
    CHAT = 3;             // 聊天服务器
    DB = 4;               // 数据库服务器
    FILE = 5;             // 文件服务器
    GATEWAY = 6;          // 网关服务器
    LOGIC = 7;            // 逻辑服务器
    LOGIN = 8;            // 登录服务器
    MATCHING = 9;         // 匹配服务器
    STORE = 10;           // 商城服务器
};

// 服务类型：客户端与服务器通信的服务类型
// 考虑到有些服务需要服务器主动发送消息，以及格式化，所以为所有服务添加响应类型
enum ServiceType
{
    // REQ_ 请求
    // RES_ 响应
    REQ_LOGIN = 0;      // 登录请求
    RES_LOGIN = 1;      // 登录响应
    REQ_LOGOUT = 2;     // 登出请求
    RES_LOGOUT = 3;     // 登出响应
    REQ_REGISTER = 4;   // 注册请求
    RES_REGISTER = 5;   // 注册响应
    REQ_HEARTBEAT = 6;  // 心跳请求
    RES_HEARTBEAT = 7;  // 心跳响应
    REQ_FILE_UPLOAD = 8;    // 文件上传请求
    RES_FILE_UPLOAD = 9;    // 文件上传响应
    REQ_FILE_DOWNLOAD = 10; // 文件下载请求
    RES_FILE_DOWNLOAD = 11; // 文件下载响应
    REQ_FILE_DELETE = 12;   // 文件删除请求
    RES_FILE_DELETE = 13;   // 文件删除响应
    REQ_FILE_LIST = 14;     // 获取文件列表请求
    RES_FILE_LIST = 15;     // 获取文件列表响应
}